src/blockdata/block.rs

```diff
@@ -155,6 +157,16 @@ impl XField {
             XField::Unknown(_, _) => 0,
         }
     }
+
+    /// Return hash of serialized XField for signing
+    pub fn signature_hash(&self) ->Result<XFieldHash, Error>  {
+        match self {
+            XField::None => Ok(XFieldHash::from_str("").unwrap()),
```

Comment:
This code is intended to return a hash of an empty string, but it actually panics at `XFieldHash::from_str("").unwrap()`. And test case `xfield_signature_hash_test_none` also seems to expect that.

If you want to generate an error when None, XFieldHash::from_str("") is not necessary. On the other hand, if you return a hash for empty data, there should be no error.

Reply:
The idea was to return an empty string as the hash. But the hash constructor rejects it after checking the length. I never realised it as it doesn't happen in any real test scenario. I'll return error and fix the unit test.